fdbserver/worker.actor.cpp

```diff
@@ -1328,20 +1329,56 @@ ACTOR Future<Void> monitorHighMemory(int64_t threshold) {
 	return Void();
 }
 
+struct StorageDiskCleaner {
+	UID id;
+	KeyValueStoreType storeType;
+	LocalityData locality;
+	std::string filename;
+	Future<Void> future;
+};
+
 struct TrackRunningStorage {
 	UID self;
 	KeyValueStoreType storeType;
+	LocalityData locality;
+	std::string filename;
 	std::set<std::pair<UID, KeyValueStoreType>>* runningStorages;
+	std::vector<StorageDiskCleaner>* storageCleaners;
```

Comment:
The way you are using `storageCleaners`, it really seems like what you want is `std::map<UID, StorageCleaner>` because
-  on insert you are manually scanning for an existing UID collision only inserting if there isn't one
-  on removal you are scanning for a UID match with std::remove_if and assuming there is only 1 match, which is true because of the pre-insert check.

Reply:
use unordered_map now. thank you!